pkg/providers/amifamily/ami.go

```diff
@@ -92,31 +98,42 @@ func (a AMIs) String() string {
 }
 
 // MapToInstanceTypes returns a map of AMIIDs that are the most recent on creationDate to compatible instancetypes
-func (a AMIs) MapToInstanceTypes(instanceTypes []*cloudprovider.InstanceType) map[string][]*cloudprovider.InstanceType {
+func MapToInstanceTypes(instanceTypes []*cloudprovider.InstanceType, amis []v1beta1.AMI) map[string][]*cloudprovider.InstanceType {
 	amiIDs := map[string][]*cloudprovider.InstanceType{}
 	for _, instanceType := range instanceTypes {
-		for _, ami := range a {
-			if err := instanceType.Requirements.Compatible(ami.Requirements, scheduling.AllowUndefinedWellKnownLabels); err == nil {
-				amiIDs[ami.AmiID] = append(amiIDs[ami.AmiID], instanceType)
+		for _, ami := range amis {
+			if err := instanceType.Requirements.Compatible(scheduling.NewNodeSelectorRequirementsWithMinValues(ami.Requirements...), scheduling.AllowUndefinedWellKnownLabels); err == nil {
+				amiIDs[ami.ID] = append(amiIDs[ami.ID], instanceType)
 				break
 			}
 		}
 	}
 	return amiIDs
 }
 
-func NewDefaultProvider(versionProvider version.Provider, ssm ssmiface.SSMAPI, ec2api ec2iface.EC2API, cache *cache.Cache) *DefaultProvider {
+func NewDefaultProvider(kubeClient client.Client, versionProvider version.Provider, ssm ssmiface.SSMAPI, ec2api ec2iface.EC2API, cache *cache.Cache) *DefaultProvider {
 	return &DefaultProvider{
 		cache:           cache,
 		ssm:             ssm,
 		ec2api:          ec2api,
+		kubeClient:      kubeClient,
 		cm:              pretty.NewChangeMonitor(),
 		versionProvider: versionProvider,
 	}
 }
 
+func (p *DefaultProvider) Get(ctx context.Context, nodeClass *v1beta1.EC2NodeClass) ([]v1beta1.AMI, error) {
+	p.RLock()
+	defer p.RUnlock()
+
+	if err := p.kubeClient.Get(ctx, types.NamespacedName{Name: nodeClass.Name}, nodeClass); err != nil {
```

Comment:
Do we really need to re-retrieve the NodeClass if we already have the nodeClass as an argument to the function?